src/stateMachine.ts

```diff
@@ -49,6 +65,21 @@ export class StateMachine<STATE, EVENT extends string | number | symbol, CALLBAC
     return this.transitions.some((trans) => (trans.fromState === this._current && trans.event === event));
   }
 
+  async getNextState(event: EVENT): Promise<STATE> {
```

Comment:
there's no reason to make it an async function. also no reason to log an error.

```
   getNextState(event: EVENT): STATE | undefined {
    const transition = this.transitions.find((tran) => tran.fromState === this._current && tran.event === event);
    return transition?.toState;
  }

```

Reply:
Fixed